File: ios/Shared/Sources/Core/Parsing/LmsCourseTitleParser.swift
import Foundation

public enum LmsCourseTitleParser {
    public static func parse(_ value: String) -> (courseCode: String?, title: String) {
        let trimmed = value.trimmingCharacters(in: .whitespacesAndNewlines)
        guard let separatorIndex = trimmed.firstIndex(of: ":") else {
            return (nil, trimmed)
        }

        let codeCandidate = trimmed[..<separatorIndex].trimmingCharacters(in: .whitespacesAndNewlines)
        let titleCandidate = trimmed[trimmed.index(after: separatorIndex)...].trimmingCharacters(
            in: .whitespacesAndNewlines)

        let courseCode = codeCandidate.isEmpty ? nil : codeCandidate
        let title = titleCandidate.isEmpty ? trimmed : titleCandidate
        return (courseCode, title)
    }
}

File: android/core/src/main/kotlin/dev/example/moodlenative/core/LmsCourseTitleParser.kt
package dev.example.moodlenative.core

data class ParsedCourseTitle(
    val courseCode: String?,
    val title: String,
)

object LmsCourseTitleParser {
    fun parse(value: String): ParsedCourseTitle {
        val trimmed = value.trim()
        val separatorIndex = trimmed.indexOf(':')
        if (separatorIndex < 0) {
            return ParsedCourseTitle(null, trimmed)
        }

        val codeCandidate = trimmed.substring(0, separatorIndex).trim()
        val titleCandidate = trimmed.substring(separatorIndex + 1).trim()
        return ParsedCourseTitle(
            courseCode = codeCandidate.ifEmpty { null },
            title = titleCandidate.ifEmpty { trimmed },
        )
    }
}
